drivers/sensor/lis3mdl/lis3mdl.c

```diff
@@ -136,11 +150,15 @@ int lis3mdl_init(const struct device *dev)
 		return -EINVAL;
 	}
 
+	/* Check if single mode should be set */
+	drv_data->single_mode = lis3mdl_odr_bits[idx] & LIS3MDL_FAST_ODR_MASK ?
+		      false : true;
+
```

Comment:
So, in this mode when fast_odr is not set it always goes in single_mode. Is this intended? I guess user may want to set continuous mode also for slower odr. I'm not able to see on the DS if this possibility exists or not.